Code/src/main/java/nl/utwente/group10/ui/components/Connection.java

```diff
@@ -175,18 +182,22 @@ public final void changed(ObservableValue<? extends Number> observable,
 
 	public final void disconnect(ConnectionAnchor anchor) {
 		if (startAnchor.isPresent() && startAnchor.get().equals(anchor)) {
-			startAnchor.get().disconnect(this);
+			startAnchor.get().disconnectFrom(this);
 			startAnchor = Optional.empty();
 		}
 		if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
-			endAnchor.get().disconnect(this);
+			endAnchor.get().disconnectFrom(this);
 			endAnchor = Optional.empty();
 		}
 	}
+	
+	public final void disconnect(Optional<? extends ConnectionAnchor> anchor){
+		disconnect(anchor.orElse(null));
+	}
 
 	public final void disconnect() {
-		disconnect(startAnchor.orElse(null));
-		disconnect(endAnchor.orElse(null));
+		disconnect(startAnchor);
+		disconnect(endAnchor);
```

Comment:
Optional<T> heeft ook een equals(), misschien heb je daar wat aan: dan hoef je volgens mij niet meer uit te pakken naar een nullable. Met andere woorden zou je in het bovenste stuk misschien

    startAnchor.isPresent() && startAnchor.get().equals(anchor)

kunnen vervangen door

    startAnchor.equals(anchor)

en de disconnect() die een ConnectionAnchor neemt kunnen vervangen door

    disconnect(Optional.of(anchor))